node/http/src/api.rs

```diff
@@ -183,4 +192,32 @@ impl HttpApi {
         };
         Ok(response)
     }
+
+    pub fn view_latest_beacon_block(&self) -> Result<SignedBeaconBlockResponse, ()> {
+        Ok(self.beacon_chain.best_block().into())
+    }
+
+    pub fn get_beacon_block_by_hash(
+        &self,
+        r: &GetBlockByHashRequest,
+    ) -> Result<SignedBeaconBlockResponse, &str> {
+        match self.beacon_chain.get_block(&BlockId::Hash(r.hash)) {
+            Some(block) => Ok(block.into()),
+            None => Err("block not found"),
+        }
+    }
+
+    pub fn view_latest_shard_block(&self) -> Result<SignedShardBlockResponse, ()> {
```

Comment:
If the error types is `()`, why don't we use `Option` instead?

Reply:
this was to maintain a consistent interface across API functions. it doesn't totally make sense now, but it will reduce a lot of boilerplate when we consolidate things in the server.